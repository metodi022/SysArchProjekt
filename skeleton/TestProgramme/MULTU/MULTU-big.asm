lui $1, 0x7000  // 32'd1879076864
ori $1, 0x7000
lui $2, 0x7fff  // 32'd2147450879
ori $2, 0x7fff
multu $1, $2    // multiply $1 and $2
mfhi $3         // 0x37ffffff
mflo $4         // 0x57ff9000

// (2^31)*((3*2^30)-1) = 1879076864 * 2147450879 = 6917529025493598208
//                     = 0x70007000 * 0x7fff7fff = 0x37ffffff57ff9000
