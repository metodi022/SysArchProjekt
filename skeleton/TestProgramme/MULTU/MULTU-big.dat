0x3c017000
0x34217000
0x3c027fff
0x34427fff
0x00220019
0x00001810
0x00002012
