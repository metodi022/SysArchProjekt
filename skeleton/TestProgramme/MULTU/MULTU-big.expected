@1 70007000
@2 7fff7fff
@3 37ffffff
@4 57ff9000
